### PR TITLE
IE10 compatibility: use css 'display' to hide/display, html5 hidden not supported

### PR DESCRIPTION
**Pull request by mrenou**
Use css 'display' to hide/display components via a custom attribut 'data-hidden'. 'hidden' html5 attribut is not supported by IE10.

### CODE REVIEW

**Comment by vlio20** [comment]
I have to say that I don't like this approach. 
1. first because it is ugly.
2. there shouldn't be styles in the variables.less file (it might be that some component won't import it) - but I am not sure where to put it...

Are there any other solutions?

**Comment by mrenou** [comment]
I didn't find other solution for browser which doesn't implement 'hidden'. I could apply css style on css selector [hidden] directly in my app and avoid update your library, but it doesn't work because the attribut 'hidden' seems to be removed from the DOM with angular code [hidden]="condition". I think it's because custom attribut must start with 'data-' with some older browsers.

Moreover, the implementation of 'hidden' by browser is exactly to add 'disaply: none', and if we use 'display: flex', hidden will not work : 
https://developer.mozilla.org/fr/docs/Web/HTML/Attributs_universels/hidden 

I'm not expert, but it seems to me that the use of a css rule handles the most cases. Even if, in my case, it's to support a very old and little used browser, but still present in some projects (pity me :smiley: ). 

Except that, the library works well on IE 10.

About styles in variables.less, I totally agree with you. In sass, we use a file named _base.scss, so I could create a file /common/styles/base.less or /common/styles/common.less. Other solution, less apreciable, we could accept some code duplication in calendar-nav.component.less and date-picker.component.less. 

Thanks for your time

**Comment by vlio20** [comment]
mmmm... what do you think about adding both `[hidden]` and `[attr.data-hidden]` + adding a note in the readme for ie10 developers. I don't believe this browser will stay for here long.

**Comment by mrenou** [comment]
Yes, I can do that.

Do you want to add the css rule on data-hidden in library (in a base.less or common.less file) ? Or let the library user does it in his app code (with a mention in the doc) ?

**Comment by vlio20** [comment]
let the library user does it in his app code (with a mention in the doc)

**Comment by vlio20** [comment]
@mrenou, I want to push this to 2.6.1
So this is the only open issue that is left.

**Comment by mrenou** [comment]
Ok I do that now

**Comment by mrenou** [comment]
Let me know if it's ok for you

File: src/app/date-picker/date-picker.component.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <div [ngClass]="{'dp-open': areCalendarsShown}">
-  <div [hidden]="componentConfig.hideInputContainer" class="dp-input-container">
+  <div [hidden]="componentConfig.hideInputContainer" [attr.data-hidden]="componentConfig.hideInputContainer" class="dp-input-container">
```
Comment (vlio20) [inline src/app/date-picker/date-picker.component.html]: break attributes to lines pls

**Comment by vlio20** [comment]
also, please make sure you do your PR on the next branch

**Comment by vlio20** [comment]
Seems like there are some conflicts with the next branch...

**Comment by mrenou** [comment]
Yep, I'll rebase on next locally, I'll inform you

**Comment by mrenou** [comment]
@vlio20 I updated commit and rebased on next

**Pull request merged**